flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -1212,23 +1208,13 @@ protected void onRegistrationFailure(final Throwable failure) {
 
 	private class JobManagerJobStatusListener implements JobStatusListener {
 
-		private volatile boolean running = true;
-
 		@Override
 		public void jobStatusChanges(
 				final JobID jobId,
 				final JobStatus newJobStatus,
 				final long timestamp,
 				final Throwable error) {
-
-			if (running) {
-				// run in rpc thread to avoid concurrency
-				runAsync(() -> jobStatusChanged(newJobStatus, timestamp, error));
-			}
-		}
-
-		private void stop() {
-			running = false;
+			jobStatusChanged(newJobStatus, timestamp, error);
```

Comment:
It's not really related to this change, but since you are touching the code: The `timestamp` and `error` parameters are not used in `jobStatusChanged`.
At least in the PR for master, we could address that?

Reply:
True, I will remove it in the master.